src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -114,59 +148,131 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                 return true;
             }
 
-            static LSP.CompletionItem CreateLSPCompletionItem(
+            static async Task<LSP.CompletionItem> CreateLSPCompletionItemAsync(
                 LSP.CompletionParams request,
+                Document document,
                 CompletionItem item,
                 long? resultId,
                 bool useVSCompletionItem,
                 CompletionTrigger completionTrigger,
-                Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>> commitCharacterRulesCache)
+                Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>> commitCharacterRulesCache,
+                CompletionService completionService,
+                string? clientName,
+                bool returnTextEdits,
+                SourceText? documentText,
+                TextSpan? defaultSpan,
+                LSP.Range? defaultRange,
+                CancellationToken cancellationToken)
             {
                 if (useVSCompletionItem)
```

Comment:
aside: i don't see why we have this option.  why wouldn't we always useVSCompletion ?

Reply:
we should definitely get rid of the roslyn completion item type - that is an outdated liveshare concept that we no longer need.  However checking for the VS client capability is correct (and returning a regular LSP completion item if not set) if we want to be a spec abiding LSP server and eventually support non-vs clients.

Reply:
wfm./